libc/src/__support/FPUtil/generic/sqrt.h

```diff
@@ -78,20 +78,14 @@ LIBC_INLINE cpp::enable_if_t<cpp::is_floating_point_v<T>, T> sqrt(T x) {
 
     FPBits_t bits(x);
 
-    if (bits.is_inf_or_nan()) {
-      if (bits.is_neg() && (bits.get_mantissa() == 0)) {
-        // sqrt(-Inf) = NaN
-        return FLT_NAN;
-      } else {
-        // sqrt(NaN) = NaN
-        // sqrt(+Inf) = +Inf
-        return x;
-      }
-    } else if (bits.is_zero()) {
+    if (bits == FPBits_t::inf(Sign::POS) || bits.is_zero()) {
+      // sqrt(+Inf) = +Inf
       // sqrt(+0) = +0
       // sqrt(-0) = -0
       return x;
-    } else if (bits.is_neg()) {
+    } else if (bits.is_inf_or_nan() || bits.is_neg()) {
```

Comment:
This will change the NaN payload when x is NaN. When x is NaN we should return it as-is.

Reply:
Is this required by the standard?
It is not explicit from the [cppreference article](https://en.cppreference.com/w/cpp/numeric/math/sqrt#:~:text=If%20the%20argument%20is%20NaN%2C%20NaN%20is%20returned.).

Reply:
The C standard (ISO/IEC 9899:201x) says
5.2.4.2.2
_A quiet NaN propagates through almost every arithmetic operation without raising a floating-point exception; a signaling NaN generally raises a floating-point exception when occurring as an arithmetic operand._

I'm still unclear about the exact semantics...

Reply:
Anyways, I think returning the same NaN makes more sense so I've changed the code accordingly.

Reply:
Yes, returning NaN as-is helps with the signaling / quiet NaN semantic, and also it follows the IEEE 754's recommendation:
```
To facilitate propagation of diagnostic information contained in NaNs,
as much of that information as possible should be preserved in NaN 
results of operations.
```